[Feature guide] Add end user's IdP information to host vitals

marko-lisica opened the pull request:
Feature guide for:
- #23236

noahtalerman commented:
Thanks @marko-lisica! haven't had a chance to review this yet. Will take a look later today or tomorrow morning.

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,77 @@
+
+_Available in Fleet Premium._
+
+Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
+host. This simplifies the process of identifying which employee is assigned to each host.
+Fleet uses System for Cross-domain Identity Management (SCIM) protocol to retreive users from IdP. This feature is available in Fleet [v4.67.0](https://github.com/fleetdm/fleet/releases/tag/fleet-v4.67.0).
```
```suggestion
Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
host. Fleet uses the System for Cross-domain Identity Management (SCIM)
```

noahtalerman replied:
Add link to SCIM

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,77 @@
+host. This simplifies the process of identifying which employee is assigned to each host.
+Fleet uses System for Cross-domain Identity Management (SCIM) protocol to retreive users from IdP. This feature is available in Fleet [v4.67.0](https://github.com/fleetdm/fleet/releases/tag/fleet-v4.67.0).
+
+Currently, Fleet supports only Okta.
```
```suggestion
```

noahtalerman replied:
If someone wants to try Entra then they can we just won't document it at first.

marko-lisica commented:
Hey @noahtalerman I added instructions for authentik. Could you take a look at it? You can use credentials from 1Password to login to authentik server, so you can check if the instructions follow everything.

I also added credentials for a test Google Workspace account.

marko-lisica commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+2. Select **Create**, then select **LDAP Source**, and select **Next**.
+3. Add friendly name (e.g. Google LDAP)
+4. Make sure that **Enable**, **Sync users** and **Sync groups** are toggled on.
+5. In **Server URL** enter `ldap://ldap.google.com`.
```
```suggestion
5. In **Server URL** enter `ldap://ldap.google.com`. For more information refer to [Google docs](https://support.google.com/a/answer/9089736?hl=en&ref_topic=9173976&sjid=5482490660946222035-EU#basic-instructions).
```

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+# Add user information from identity provider (IdP) to host
```
```suggestion
# Foreign host vitals: identity provider (IdP) groups
```

Thoughts on this title? Mike and I have been using "Foreign vitals" a lot. I think it's a good product marketing approach. We might add more foreign vitals later

noahtalerman replied:
If that new title looks good, I think let's change the file name too so that it's used as the website URL. The website uses the filename to generate the guide's URL.

marko-lisica replied:
@noahtalerman I changed `Foreign host vitals: identity provider (IdP) groups` to `Foreign host vitals: identity provider (IdP) user information`, since we add more than groups. Wdyt?

noahtalerman replied:
@marko-lisica I think let's go with something like "Foreign vitals: map users from your IdP to hosts"

We have "foreign vitals", mention IdP (I think we keep it as an acronym to keep the title short, and we have mapping users to hosts.

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+
+_Available in Fleet Premium._
+
+Learn how to connect your identity provider (IdP) to retrieve end-user information and map it to a
+host. Fleet uses [SCIM](https://scim.cloud/) protocol.
+
+Below you can learn how to connect [Okta](https://www.okta.com/), [Microsoft Entra ID](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id) or [Authentik](https://goauthentik.io/).
```
```suggestion
Fleet supports [Okta](#okta), [authentik](#authentik), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.

To add IdP host vitals, follow steps for your IdP.
```

I think we can make this language consistent with what we have here in the [SSO docs](https://fleetdm.com/docs/deploy/single-sign-on-sso#single-sign-on-sso):

![Screenshot 2025-03-21 at 9 35 56 AM](https://github.com/user-attachments/assets/3b3526a5-ecf0-4217-9603-18adcf076bbb)

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+Below you can learn how to connect [Okta](https://www.okta.com/), [Microsoft Entra ID](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id) or [Authentik](https://goauthentik.io/).
+
+
+## Connect Okta
```
```suggestion
## Okta
```

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+   Fleet, and Fleet will map those groups to already provisioned users .
+
+
+## Connect Microsoft Entra ID
```
```suggestion
## Microsoft Entra ID
```

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+> requests are triggered every 20-40 minutes. If you edit user information in Entra, it might
+> take up to 40 minutes to see the change in Fleet.
+
+## Connect Google Workspace (with authentik)
```
```suggestion
## Google Workspace
```

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
```diff
@@ -0,0 +1,284 @@
+
+## Connect Google Workspace (with authentik)
+
+Google Workspace doesn't support SCIM provisioning, so we suggest using [authentik](https://goauthentik.io/) to import users from Google Workspace and then add users to Fleet via SCIM protocol.
```
```suggestion
Google Workspace doesn't natively support the [SCIM](https://scim.cloud/) standard. The best practice is to import users to [authentik](https://goauthentik.io/). Authentik then adds users to Fleet.
```

noahtalerman replied:
"Best practice" instead of "suggest" to help folks feel confident.

noahtalerman commented on articles/add-user-info-from-idp-to-host.md:
````diff
@@ -0,0 +1,284 @@
+            "ldap_uniq": ldap.get("objectSid"),
+        },
+    }
+    ```
````
@marko-lisica looks like you might be missing a closing "```" here. The markdown formatting looks off.

marko-lisica replied:
@noahtalerman Seems as GitHub bug. I tested and it works well when I run website.

<img width="846" alt="Screenshot 2025-03-21 at 18 19 54" src="https://github.com/user-attachments/assets/2a1ee95f-1668-4c03-944a-719d3c7a7952" />
<img width="834" alt="Screenshot 2025-03-21 at 18 20 01" src="https://github.com/user-attachments/assets/f0d8a3b5-9bfc-46bf-9bdd-d71d25716a4a" />

noahtalerman reviewed (changes requested):
Looking good! I dropped some initial feedback.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+
+_Available in Fleet Premium._
+
+Fleet supports [Okta](#okta), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), [Google Workspace](#google-workspace), [authentik](#google-workspace),as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.
```
```suggestion
Fleet supports [Okta](#okta), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), [Google Workspace](#google-workspace), [authentik](#google-workspace), as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+
+_Available in Fleet Premium._
+
+Fleet supports [Okta](#okta), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), [Google Workspace](#google-workspace), [authentik](#google-workspace), as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.
+
+To add IdP host vitals, follow steps for your IdP.
```
```suggestion
To add IdP host vitals, like the end user's groups and full name, follow steps for your IdP. Fleet supports [Okta](#okta), [Microsoft Active Directory (AD) / Entra ID](#microsoft-entra-id), [Google Workspace](#google-workspace), [authentik](#google-workspace), as well as any other identity provider (IdP) that supports the [SCIM](https://scim.cloud/) standard.
```

After reading this again, I think we swap sentences to start with what the user wants to do. Add end user info.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+
+To connect Okta to Fleet, follow these steps:
+
+1. Head to Okta admin dashboard
```
```suggestion
1. Head to Okta admin dashboard.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+To connect Okta to Fleet, follow these steps:
+
+1. Head to Okta admin dashboard.
+2. Select **Applications > Applications** in the main menu, then select **Create App Integration**.
```
```suggestion
2. In the main menu, select **Applications > Applications**, then select **Create App Integration**.
```

I think it's a good idea to start with the menu here. First go to the menu, then...

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+1. Head to Okta admin dashboard.
+2. Select **Applications > Applications** in the main menu, then select **Create App Integration**.
+3. Select **SAML 2.0** option and select **Next**.
+4. On **General Settings** page, add friendly **App name** (e.g Fleet SCIM), and select **Next**.
```
```suggestion
4. On the **General Settings** page, add a friendly **App name** (e.g Fleet SCIM), and select **Next**.
```

Articles here.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+2. Select **Applications > Applications** in the main menu, then select **Create App Integration**.
+3. Select **SAML 2.0** option and select **Next**.
+4. On **General Settings** page, add friendly **App name** (e.g Fleet SCIM), and select **Next**.
+5. On **SAML Settings** page, add any URL to **Single sign-on URL** and **Audience URI (SP Entity ID)** fields, and select **Next**.
```
```suggestion
5. On the **SAML Settings** page, add any URL to the **Single sign-on URL** and **Audience URI (SP Entity ID)** fields, and select **Next**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+3. Select **SAML 2.0** option and select **Next**.
+4. On **General Settings** page, add friendly **App name** (e.g Fleet SCIM), and select **Next**.
+5. On **SAML Settings** page, add any URL to **Single sign-on URL** and **Audience URI (SP Entity ID)** fields, and select **Next**.
+>Okta requires to setup SAML settings in order to setup SCIM integration. Since we don't need SSO, URL can be anything.
```
```suggestion
> Okta requires us to setup SAML settings in order to setup a SCIM integration. Since we don't need SAML right now, you can set the URL to anything like "example.fleetdm.com".
```

We can suggest a URL right?

marko-lisica replied:
Yes, we can. Can be anything.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+4. On **General Settings** page, add friendly **App name** (e.g Fleet SCIM), and select **Next**.
+5. On **SAML Settings** page, add any URL to **Single sign-on URL** and **Audience URI (SP Entity ID)** fields, and select **Next**.
+>Okta requires to setup SAML settings in order to setup SCIM integration. Since we don't need SSO, URL can be anything.
+6. On **Feedback** page, provide feedback if you want, and select **Finish**.
```
```suggestion
6. On the **Feedback** page, provide feedback if you want, and select **Finish**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+5. On **SAML Settings** page, add any URL to **Single sign-on URL** and **Audience URI (SP Entity ID)** fields, and select **Next**.
+>Okta requires to setup SAML settings in order to setup SCIM integration. Since we don't need SSO, URL can be anything.
+6. On **Feedback** page, provide feedback if you want, and select **Finish**.
+7. Select **General** tab of the newly created app, then select **Edit** in **App Settings**.
```
```suggestion
7. Select the **General** tab in your newly created app and then select **Edit** in **App Settings**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+6. On **Feedback** page, provide feedback if you want, and select **Finish**.
+7. Select **General** tab of the newly created app, then select **Edit** in **App Settings**.
+8. For **Provisioning**, select **SCIM**, and select **Save**.
+9. Select **Provisioning** tab, then in **SCIM Connection**, select **Edit**.
```
```suggestion
9. Select the **Provisioning** tab, then in **SCIM Connection**, select **Edit**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+>Okta requires to setup SAML settings in order to setup SCIM integration. Since we don't need SSO, URL can be anything.
+6. On the **Feedback** page, provide feedback if you want, and select **Finish**.
+7. Select the **General** tab in your newly created app and then select **Edit** in **App Settings**.
+8. For **Provisioning**, select **SCIM**, and select **Save**.
```
```suggestion
8. For **Provisioning**, select **SCIM** and select **Save**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+7. Select the **General** tab in your newly created app and then select **Edit** in **App Settings**.
+8. For **Provisioning**, select **SCIM** and select **Save**.
+9. Select the **Provisioning** tab and then, in **SCIM Connection**, select **Edit**.
+10. In **SCIM connector base URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
```
```suggestion
10. For the **SCIM connector base URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+8. For **Provisioning**, select **SCIM** and select **Save**.
+9. Select the **Provisioning** tab and then, in **SCIM Connection**, select **Edit**.
+10. For the **SCIM connector base URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
+11. In **Unique identifier field for users**, enter `userName`.
```
```suggestion
11. For the **Unique identifier field for users**, enter `userName`.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+9. Select the **Provisioning** tab and then, in **SCIM Connection**, select **Edit**.
+10. For the **SCIM connector base URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
+11. In **Unique identifier field for users**, enter `userName`.
+12. For **Supported provisioning actions**, select **Push New Users**, **Push Profile Updates**, and **Push Groups**.
+13. For **Authentication Mode** select **HTTP Header**.
```
```suggestion
12. For the **Supported provisioning actions**, select **Push New Users**, **Push Profile Updates**, and **Push Groups**.
13. For the **Authentication Mode**, select **HTTP Header**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+11. In **Unique identifier field for users**, enter `userName`.
+12. For **Supported provisioning actions**, select **Push New Users**, **Push Profile Updates**, and **Push Groups**.
+13. For **Authentication Mode** select **HTTP Header**.
+14. Create Fleet API-only user with maintainer permissions, copy API token for that user, and paste it to Okta, in **Authorization** field.
```
```suggestion
14. Create a Fleet API-only user with maintainer permissions and copy API token for that user (learn how [here](https://fleetdm.com/guides/fleetctl#create-api-only-user)). Paste your API token in Okta's **Authorization** field.
```

noahtalerman replied:
I think let's help the user create an API-only user.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,283 @@
+12. For **Supported provisioning actions**, select **Push New Users**, **Push Profile Updates**, and **Push Groups**.
+13. For **Authentication Mode** select **HTTP Header**.
+14. Create Fleet API-only user with maintainer permissions, copy API token for that user, and paste it to Okta, in **Authorization** field.
+15. Select **Test Connector Configuration** button. You should see success message in Okta.
+16. Head to Fleet, select **Settings > Integrations > Identity provider (IdP)** and verify that Fleet successfully received the request from IdP.
+17. Back in Okta, select **Save**
+18. Under **Provisioning** tab, select **To App**, then select **Edit** in **Provisioning to App** section. Enable **Create Users**, **Update User Attributes**, and **Deactivate Users**, then select **Save**.
+19. On the same page, make sure that `givenName` and `familyName` have Okta value assigned to it.
+    Currently, Fleet support `userName`, `givenName`, and `familyName` SCIM attributes and they are
+    required as well. Delete the rest of the attributes.
```
```suggestion
15. Select the **Test Connector Configuration** button. You should see success message in Okta.
16. In Fleet, head to **Settings > Integrations > Identity provider (IdP)** and verify that Fleet successfully received the request from IdP.
17. Back in Okta, select **Save**.
18. Under the **Provisioning** tab, select **To App** and then select **Edit** in the **Provisioning to App** section. Enable **Create Users**, **Update User Attributes**, **Deactivate Users**, and then select **Save**.
19. On the same page, make sure that `givenName` and `familyName` have Okta value assigned to it. Currently, Fleet requires the `userName`, `givenName`, and `familyName` SCIM attributes. Delete the rest of the attributes.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,281 @@
+
+### Assign users and groups to Fleet
+
+To send users and groups information to Fleet, you have to assign them to SCIM app that you created previously.
```
```suggestion
To send users and groups information to Fleet, you have to assign them to your new SCIM app.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,281 @@
+
+To send users and groups information to Fleet, you have to assign them to your new SCIM app.
+
+1. Select **Directory > Groups** in the main menu, then select **Add group**. Name it so you know that users from this group will be provisioned to Fleet (e.g "Fleet human-device mapping").
+2. On the same page, select **Rules** tab. Create rule that will assign users from groups that you want to provision to Fleet to newly created "Fleet human-device mapping" group.
+![Okta group rule](../website/assets/images/articles/okta-scim-group-rules.png)
+3. Select **Applications > Applications** in the main menu, select app that you created previously, then select **Assignements** tab.
+4. Select **Assign > Assign to Groups**, then click **Assign** next to the "Fleet human-device mapping" group, then select **Done**. Now all users that you assigned to "Fleet human-device mapping" group via rule will be provisioned to Fleet. It may take a while if you have many users.
+5. On the same page, select **Push Groups** tab, then select **Push Groups > Find groups by name**,
+   and add all groups that you assigned via rule to "Fleet human-device mapping" group previously (make sure that
+   **Push group memberships immediately** is selected). All groups will be provisioned to
+   Fleet, and Fleet will map those groups to already provisioned users .
```
```suggestion
1. In OKta's main menu **Directory > Groups** and then select **Add group**. Name it "Fleet human-device mapping".
2. On the same page, select the **Rules** tab. Create a rule that will assign users to your  "Fleet human-device mapping" group.
![Okta group rule](../website/assets/images/articles/okta-scim-group-rules.png)
3. In the main menu, select **Applications > Applications**  and select your new SCIM app. Then, select the **Assignments** tab.
4. Select **Assign > Assign to Groups** and then select **Assign** next to the "Fleet human-device mapping" group. Then, select **Done**. Now all users that you assigned to the  "Fleet human-device mapping" group will be provisioned to Fleet. It may take a while if you have many users.
5. On the same page, select **Push Groups** tab. Then, select **Push Groups > Find groups by name** and add all groups that you assigned to "Fleet human-device mapping" group previously (make sure that **Push group memberships immediately** is selected). All groups will be provisioned in Fleet, and Fleet will map those groups to users.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,278 @@
+2. On the same page, select the **Rules** tab. Create a rule that will assign users to your  "Fleet human-device mapping" group.
+![Okta group rule](../website/assets/images/articles/okta-scim-group-rules.png)
+3. In the main menu, select **Applications > Applications**  and select your new SCIM app. Then, select the **Assignments** tab.
+4. Select **Assign > Assign to Groups** and then select **Assign** next to the "Fleet human-device mapping" group. Then, select **Done**. Now all users that you assigned to the  "Fleet human-device mapping" group will be provisioned to Fleet. It may take a while if you have many users.
```
> It may take a while if you have many users.

How long? 1 hour? Can we give the user a more specific estimate here?

marko-lisica replied:
@noahtalerman I'm not sure. I assumed this and wanted to mention it because I think it won't be instant if there are, for example, 10,000 users. We could omit this and find out how much time it actually takes. If it's too long, we can add that to the guide.

noahtalerman replied:
@marko-lisica when you tested Okta with only a couple users how fast was it?

If it was quick I think we can omit this and add it back when we learn from our customers that it's slower at scale.

marko-lisica replied:
It was almost instant for 5 users. I removed it for now.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,278 @@
+
+To connect Entra ID to Fleet, follow these steps:
+
+1. Head to [Microsoft Entra admin](https://entra.microsoft.com/).
+2. Select **Applications > Enterprise applications** in the main menu, select **+ New
+   application**, then select **+Create your own application**.
+3. Add a friendly name of the app (e.g Fleet SCIM), select **Integrate any other application you
+   don't find in the gallery (Non-gallery)**, then select **Create**.
+4. Select **Provisioning** from the side menu.
+5. In **Get started with application provisioning** section, select **Connect your application**.
+6. In **Tenant URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
+7. Create Fleet API-only user with maintainer permissions, copy API token for that user, and paste
+   it in **Secret token** field.
+8. Select **Test connection** button. You should see success message.
+9. Select **Create** and after successfull creation, you'll be redirected to the overview page.
+10. Select **Attribute mapping** from the side menu, then select **Provision Microsoft Entra ID
+    Groups**.
+11. Ensure that the attributes `displayName`, `members`, and `externalId` are
+    mapped to **Microsoft Entra ID Attribute**. Currently, Fleet support only these attributes and they are required as well. Delete the rest of the attributes, select **Save**, and after it's
+    saved, select close icon on the top right corner.
+![Entra SCIM attributes mapping for groups](../website/assets/images/articles/entra-group-scim-attributes.png)    
+12. Select **Provision Microsoft Entra ID Users**.
+13. Ensure that the attributes `userName`, `givenName`, `familyName`, `active`, and `externalId` are
+    mapped to **Microsoft Entra ID Attribute**. Currently, Fleet support only these attributes and
+    they are required as well. Delete the rest of the attributes, select **Save**, and after it's
+    saved, select close icon on the top right corner.
+![Entra SCIM attributes mapping for users](../website/assets/images/articles/entra-user-scim-attributes.png)  
+14. Select **Users and groups** from the side menu, select **+ Add user/group**, then select
+    **None Selected** link.
+    
+15. Select users and groups that you want to add to Fleet, then select
+    **Assign**. 
+16. Select **Overview** from the side menu and select **Start provisioning**.
```
```suggestion
1. Head to [Microsoft Entra](https://entra.microsoft.com/).
2. In the main menu, select **Applications > Enterprise applications**. Then, select **+ New
   application** and **+Create your own application**.
3. Add a friendly name for the app (e.g. Fleet SCIM), select **Integrate any other application you don't find in the gallery (Non-gallery)**, and select **Create**.
4. From the side menu, select **Provisioning**.
5. In **Get started with application provisioning** section, select **Connect your application**.
6. For the **Tenant URL**, enter `https://<your_fleet_server_url>/api/v1/fleet/scim`.
7. Create a Fleet API-only user with maintainer permissions and copy API token for that user (learn how [here](https://fleetdm.com/guides/fleetctl#create-api-only-user)). Paste your API token in the **Secret token** field.
8. Select the **Test connection** button. You should see success message.
9. Select **Create** and, after successful creation, you'll be redirected to the overview page.
10. From the side menu, select **Attribute mapping** and then select **Provision Microsoft Entra ID Groups**.
11. Ensure that the attributes `displayName`, `members`, and `externalId` are mapped to **Microsoft Entra ID Attribute**. Currently, Fleet support only these attributes and they are required as well. Delete the rest of the attributes, select **Save**, and after it's
    saved, select close icon on the top right corner.
![Entra SCIM attributes mapping for groups](../website/assets/images/articles/entra-group-scim-attributes.png)    
12. Select **Provision Microsoft Entra ID Users**.
13. Ensure that the attributes `userName`, `givenName`, `familyName`, `active`, and `externalId` are
    mapped to **Microsoft Entra ID Attribute**. Currently, Fleet requires the `userName`, `givenName`, and `familyName` SCIM attributes. Delete the rest of the attributes. Then, elect **Save** and select the close icon on the top right corner.
![Entra SCIM attributes mapping for users](../website/assets/images/articles/entra-user-scim-attributes.png)  
14. Next, from the side menu, select **Users and groups** , **+ Add user/group**, and **None Selected**.
15. Select the users and groups that you want to add to Fleet and then select **Assign**. 
16. From the side menu, select **Overview** and select **Start provisioning**.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,269 @@
+    mapped to **Microsoft Entra ID Attribute**. Currently, Fleet requires the `userName`, `givenName`, and `familyName` SCIM attributes. Delete the rest of the attributes. Then, elect **Save** and select the close icon on the top right corner.
+![Entra SCIM attributes mapping for users](../website/assets/images/articles/entra-user-scim-attributes.png)  
+14. Next, from the side menu, select **Users and groups** , **+ Add user/group**, and **None Selected**.
+15. Select the users and groups that you want to add to Fleet and then select **Assign**. 
```
```suggestion
15. Select the users and groups that you want to map to hosts in Fleet and then select **Assign**. 
```

I think we want to be explicit about "host mapping" in this guide. Instead of just saying "add users and groups". 

Why? We already know there's going to be confusion around Fleet not having a users table. Fleet is doing human-device mapping.

marko-lisica replied:
Makes sense.

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,269 @@
+15. Select the users and groups that you want to add to Fleet and then select **Assign**. 
+16. From the side menu, select **Overview** and select **Start provisioning**.
+
+> After you started provisioning Microsoft Entra ID sends data to Fleet immediately. Subsequent
+> requests are triggered every 20-40 minutes. If you edit user information in Entra, it might
+> take up to 40 minutes to see the change in Fleet.
```
```suggestion
Now, Microsoft Entra ID will send data to Fleet immediately. Future requests happen every 20-40 minutes. So, if you edit user information in Entra, it might take up to 40 minutes to see the change in Fleet.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,267 @@
+
+## Google Workspace
+
+Google Workspace doesn't natively support the [SCIM](https://scim.cloud/) standard. The best practice is to import users to [authentik](https://goauthentik.io/). Authentik then adds users to Fleet.
```
```suggestion
Google Workspace doesn't natively support the [SCIM](https://scim.cloud/) standard. The best practice is to export users to [authentik](https://goauthentik.io/). Authentik then adds users to Fleet.
```

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,261 @@
+To verify that user information is added to a hosts, go to the host that has IdP email assigned, and
+verify that **Full name (IdP)** and **Groups (IdP)** are populated correctly.
+
+### Troubleshoot errors
```
```suggestion
### Troubleshooting
```

What do you think?

noahtalerman commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,261 @@
+10. After few minutes, you should see users mapped to hosts in Fleet.
+
+
+## Verify connection in Fleet
```
```suggestion
## Verify connection
```

marko-lisica commented on articles/foreign-host-vitals-identity-provider-idp-user-information.md:
```diff
@@ -0,0 +1,261 @@
+2. On the same page, select the **Rules** tab. Create a rule that will assign users to your  "Fleet human-device mapping" group.
+![Okta group rule](../website/assets/images/articles/okta-scim-group-rules.png)
+3. In the main menu, select **Applications > Applications**  and select your new SCIM app. Then, select the **Assignments** tab.
+4. Select **Assign > Assign to Groups** and then select **Assign** next to the "Fleet human-device mapping" group. Then, select **Done**. Now all users that you assigned to the  "Fleet human-device mapping" group will be provisioned to Fleet. It may take a while if you have many users.
```
```suggestion
4. Select **Assign > Assign to Groups** and then select **Assign** next to the "Fleet human-device mapping" group. Then, select **Done**. Now all users that you assigned to the  "Fleet human-device mapping" group will be provisioned to Fleet.
```

marko-lisica commented:
> Looking good! Made some tweaks and suggested some changes (below).
> 
> Also, the Okta and Entra ID sections are long. 19 and 16 steps. Can we break them up into smaller chunks like you did for authentik? That way, the user doesn't get discouraged or lose what step they're on.

@noahtalerman Thanks for the feedback! I already broke Okta into 2 sections, SCIM app creation, (which I don't think it makes sense to break into more sections) and user/groups assignment.

I could try to merge some steps to cut at least few of them to make it shorter. Wdyt?

For Entra, it's part of the one flow, and I don't think it makes sense to have more sections. I could try to merge few steps here as well.

The only reason why I broke Google Workspace and authentik integration, is because instructions cover 2 tools and in authentic you have to do 2 different configurations (LDAP "import" and SCIM "export").

noahtalerman reviewed (approved):
Sweet!
